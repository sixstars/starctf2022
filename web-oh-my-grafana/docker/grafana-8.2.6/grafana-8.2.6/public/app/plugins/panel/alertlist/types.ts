import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';

export enum SortOrder {
  AlphaAsc = 1,
  AlphaDesc,
  Importance,
  TimeAsc,
  TimeDesc,
}

export enum ShowOption {
  Current = 'current',
  RecentChanges = 'changes',
}

export interface AlertListOptions {
  showOptions: ShowOption;
  maxItems: number;
  sortOrder: SortOrder;
  dashboardAlerts: boolean;
  alertName: string;
  dashboardTitle: string;
  tags: string[];
  stateFilter: {
    ok: boolean;
    paused: boolean;
    no_data: boolean;
    execution_error: boolean;
    alerting: boolean;
    pending: boolean;
  };
  folderId: number;
}

export interface UnifiedAlertListOptions {
  maxItems: number;
  sortOrder: SortOrder;
  dashboardAlerts: boolean;
  alertName: string;
  showInstances: boolean;
  folder: { id: number; title: string };
  stateFilter: {
    [K in PromAlertingRuleState]: boolean;
  };
}
