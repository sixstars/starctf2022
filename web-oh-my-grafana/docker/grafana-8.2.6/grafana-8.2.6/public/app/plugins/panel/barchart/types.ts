import {
  OptionsWithLegend,
  OptionsWithTextFormatting,
  OptionsWithTooltip,
  AxisConfig,
  VisibilityMode,
  GraphGradientMode,
  HideableFieldConfig,
  StackingMode,
} from '@grafana/schema';
import { VizOrientation } from '@grafana/data';

/**
 * @alpha
 */
export interface BarChartOptions extends OptionsWithLegend, OptionsWithTooltip, OptionsWithTextFormatting {
  orientation: VizOrientation;
  stacking: StackingMode;
  showValue: VisibilityMode;
  barWidth: number;
  groupWidth: number;
  rawValue: (seriesIdx: number, valueIdx: number) => number;
}

/**
 * @alpha
 */
export interface BarChartFieldConfig extends AxisConfig, HideableFieldConfig {
  lineWidth?: number; // 0
  fillOpacity?: number; // 100
  gradientMode?: GraphGradientMode;
}

/**
 * @alpha
 */
export const defaultBarChartFieldConfig: BarChartFieldConfig = {
  lineWidth: 1,
  fillOpacity: 80,
  gradientMode: GraphGradientMode.None,
  axisSoftMin: 0,
};
