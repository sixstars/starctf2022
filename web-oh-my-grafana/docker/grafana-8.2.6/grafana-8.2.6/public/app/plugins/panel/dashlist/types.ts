export interface DashListOptions {
  showStarred: boolean;
  showRecentlyViewed: boolean;
  showSearch: boolean;
  showHeadings: boolean;
  maxItems: number;
  query: string;
  folderId: number;
  tags: string[];
}
