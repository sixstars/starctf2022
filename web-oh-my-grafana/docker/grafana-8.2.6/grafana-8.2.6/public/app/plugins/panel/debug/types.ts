export type UpdateConfig = {
  [K in keyof UpdateCounters]: boolean;
};

export type UpdateCounters = {
  render: number;
  dataChanged: number;
  schemaChanged: number;
};

export enum DebugMode {
  Render = 'render',
  Events = 'events',
  Cursor = 'cursor',
}

export interface DebugPanelOptions {
  mode: DebugMode;
  counters?: UpdateConfig;
}
