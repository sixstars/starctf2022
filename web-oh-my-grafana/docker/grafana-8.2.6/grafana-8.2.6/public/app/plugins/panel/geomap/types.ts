import { MapLayerOptions } from '@grafana/data';
import { Units } from 'ol/proj/Units';
import { MapCenterID } from './view';

export interface ControlsOptions {
  // Zoom (upper left)
  showZoom?: boolean;

  // let the mouse wheel zoom
  mouseWheelZoom?: boolean;

  // Lower right
  showAttribution?: boolean;

  // Scale options
  showScale?: boolean;
  scaleUnits?: Units;

  // Show debug
  showDebug?: boolean;
}

export interface MapViewConfig {
  id: string; // placename > lookup
  lat?: number;
  lon?: number;
  zoom?: number;
  minZoom?: number;
  maxZoom?: number;
  shared?: boolean;
}

export const defaultView: MapViewConfig = {
  id: MapCenterID.Zero,
  lat: 0,
  lon: 0,
  zoom: 1,
};

export interface GeomapPanelOptions {
  view: MapViewConfig;
  controls: ControlsOptions;
  basemap: MapLayerOptions;
  layers: MapLayerOptions[];
}
