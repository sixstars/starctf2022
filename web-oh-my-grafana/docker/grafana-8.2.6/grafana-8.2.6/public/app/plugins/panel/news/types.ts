export interface NewsItem {
  date: number;
  title: string;
  link: string;
  content: string;
  ogImage?: string | null;
}

/**
 * Helper class for rss-parser
 */
export interface RssFeed {
  title?: string;
  description?: string;
  items: RssItem[];
}

export interface RssItem {
  title: string;
  link: string;
  pubDate?: string;
  content?: string;
  contentSnippet?: string;
  ogImage?: string | null;
}
