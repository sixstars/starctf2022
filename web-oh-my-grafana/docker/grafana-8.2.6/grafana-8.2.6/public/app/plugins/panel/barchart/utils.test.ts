import { prepareGraphableFrames, preparePlotConfigBuilder, preparePlotFrame } from './utils';
import {
  LegendDisplayMode,
  TooltipDisplayMode,
  VisibilityMode,
  GraphGradientMode,
  StackingMode,
} from '@grafana/schema';
import {
  createTheme,
  DefaultTimeZone,
  EventBusSrv,
  FieldConfig,
  FieldType,
  getDefaultTimeRange,
  MutableDataFrame,
  VizOrientation,
} from '@grafana/data';
import { BarChartFieldConfig, BarChartOptions } from './types';

function mockDataFrame() {
  const df1 = new MutableDataFrame({
    refId: 'A',
    fields: [{ name: 'ts', type: FieldType.string, values: ['a', 'b', 'c'] }],
  });

  const df2 = new MutableDataFrame({
    refId: 'B',
    fields: [{ name: 'ts', type: FieldType.time, values: [1, 2, 4] }],
  });

  const f1Config: FieldConfig<BarChartFieldConfig> = {
    displayName: 'Metric 1',
    decimals: 2,
    unit: 'm/s',
    custom: {
      gradientMode: GraphGradientMode.Opacity,
      lineWidth: 2,
      fillOpacity: 0.1,
    },
  };

  const f2Config: FieldConfig<BarChartFieldConfig> = {
    displayName: 'Metric 2',
    decimals: 2,
    unit: 'kWh',
    custom: {
      gradientMode: GraphGradientMode.Hue,
      lineWidth: 2,
      fillOpacity: 0.1,
    },
  };

  df1.addField({
    name: 'metric1',
    type: FieldType.number,
    config: f1Config,
    state: {},
  });

  df2.addField({
    name: 'metric2',
    type: FieldType.number,
    config: f2Config,
    state: {},
  });

  return preparePlotFrame([df1, df2]);
}

jest.mock('@grafana/data', () => ({
  ...(jest.requireActual('@grafana/data') as any),
  DefaultTimeZone: 'utc',
}));

describe('BarChart utils', () => {
  describe('preparePlotConfigBuilder', () => {
    const frame = mockDataFrame();

    const config: BarChartOptions = {
      orientation: VizOrientation.Auto,
      groupWidth: 20,
      barWidth: 2,
      showValue: VisibilityMode.Always,
      legend: {
        displayMode: LegendDisplayMode.List,
        placement: 'bottom',
        calcs: [],
      },
      stacking: StackingMode.None,
      tooltip: {
        mode: TooltipDisplayMode.None,
      },
      text: {
        valueSize: 10,
      },
      rawValue: (seriesIdx: number, valueIdx: number) => frame.fields[seriesIdx].values.get(valueIdx),
    };

    it.each([VizOrientation.Auto, VizOrientation.Horizontal, VizOrientation.Vertical])('orientation', (v) => {
      const result = preparePlotConfigBuilder({
        ...config,
        orientation: v,
        frame: frame!,
        theme: createTheme(),
        timeZone: DefaultTimeZone,
        getTimeRange: getDefaultTimeRange,
        eventBus: new EventBusSrv(),
        allFrames: [frame],
      }).getConfig();
      expect(result).toMatchSnapshot();
    });

    it.each([VisibilityMode.Always, VisibilityMode.Auto])('value visibility', (v) => {
      expect(
        preparePlotConfigBuilder({
          ...config,
          showValue: v,
          frame: frame!,
          theme: createTheme(),
          timeZone: DefaultTimeZone,
          getTimeRange: getDefaultTimeRange,
          eventBus: new EventBusSrv(),
          allFrames: [frame],
        }).getConfig()
      ).toMatchSnapshot();
    });

    it.each([StackingMode.None, StackingMode.Percent, StackingMode.Normal])('stacking', (v) => {
      expect(
        preparePlotConfigBuilder({
          ...config,
          stacking: v,
          frame: frame!,
          theme: createTheme(),
          timeZone: DefaultTimeZone,
          getTimeRange: getDefaultTimeRange,
          eventBus: new EventBusSrv(),
          allFrames: [frame],
        }).getConfig()
      ).toMatchSnapshot();
    });
  });

  describe('prepareGraphableFrames', () => {
    it('will warn when there is no data in the response', () => {
      const result = prepareGraphableFrames([], createTheme(), StackingMode.None);
      expect(result.warn).toEqual('No data in response');
    });

    it('will warn when there is no string field in the response', () => {
      const df = new MutableDataFrame({
        fields: [
          { name: 'a', type: FieldType.time, values: [1, 2, 3, 4, 5] },
          { name: 'value', values: [1, 2, 3, 4, 5] },
        ],
      });
      const result = prepareGraphableFrames([df], createTheme(), StackingMode.None);
      expect(result.warn).toEqual('Bar charts requires a string field');
      expect(result.frames).toBeUndefined();
    });

    it('will warn when there are no numeric fields in the response', () => {
      const df = new MutableDataFrame({
        fields: [
          { name: 'a', type: FieldType.string, values: ['a', 'b', 'c', 'd', 'e'] },
          { name: 'value', type: FieldType.boolean, values: [true, true, true, true, true] },
        ],
      });
      const result = prepareGraphableFrames([df], createTheme(), StackingMode.None);
      expect(result.warn).toEqual('No numeric fields found');
      expect(result.frames).toBeUndefined();
    });

    it('will convert NaN and Infinty to nulls', () => {
      const df = new MutableDataFrame({
        fields: [
          { name: 'a', type: FieldType.string, values: ['a', 'b', 'c', 'd', 'e'] },
          { name: 'value', values: [-10, NaN, 10, -Infinity, +Infinity] },
        ],
      });
      const result = prepareGraphableFrames([df], createTheme(), StackingMode.None);

      const field = result.frames![0].fields[1];
      expect(field!.values.toArray()).toMatchInlineSnapshot(`
      Array [
        -10,
        null,
        10,
        null,
        null,
      ]
    `);
    });
  });
});
