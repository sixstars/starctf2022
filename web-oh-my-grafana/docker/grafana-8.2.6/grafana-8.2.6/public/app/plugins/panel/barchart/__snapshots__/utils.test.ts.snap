// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart utils preparePlotConfigBuilder orientation 1`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder orientation 2`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder orientation 3`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": 1,
      "distr": 2,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder stacking 1`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder stacking 2`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder stacking 3`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder value visibility 1`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;

exports[`BarChart utils preparePlotConfigBuilder value visibility 2`] = `
Object {
  "axes": Array [
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 15,
      "grid": Object {
        "show": false,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "x",
      "show": true,
      "side": 3,
      "size": [Function],
      "space": [Function],
      "splits": [Function],
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": false,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
    Object {
      "font": "12px \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
      "gap": 5,
      "grid": Object {
        "show": true,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "labelGap": 0,
      "scale": "m/s",
      "show": true,
      "side": 2,
      "size": [Function],
      "space": [Function],
      "splits": undefined,
      "stroke": "rgb(204, 204, 220)",
      "ticks": Object {
        "show": true,
        "size": 4,
        "stroke": "rgba(240, 250, 255, 0.09)",
        "width": 1,
      },
      "timeZone": undefined,
      "values": [Function],
    },
  ],
  "cursor": Object {
    "drag": Object {
      "setScale": false,
    },
    "focus": Object {
      "prox": 30,
    },
    "points": Object {
      "fill": [Function],
      "show": false,
      "size": [Function],
      "stroke": [Function],
      "width": [Function],
    },
    "x": false,
    "y": false,
  },
  "hooks": Object {
    "draw": Array [
      [Function],
    ],
    "drawClear": Array [
      [Function],
    ],
    "init": Array [
      [Function],
    ],
  },
  "mode": 1,
  "padding": undefined,
  "scales": Object {
    "m/s": Object {
      "auto": true,
      "dir": 1,
      "distr": 1,
      "log": undefined,
      "ori": 0,
      "range": [Function],
      "time": undefined,
    },
    "x": Object {
      "auto": true,
      "dir": -1,
      "distr": 2,
      "log": undefined,
      "ori": 1,
      "range": [Function],
      "time": false,
    },
  },
  "select": undefined,
  "series": Array [
    Object {
      "value": [Function],
    },
    Object {
      "facets": undefined,
      "fill": [Function],
      "paths": [Function],
      "points": Object {
        "fill": "#808080",
        "filter": undefined,
        "size": undefined,
        "stroke": "#808080",
      },
      "pxAlign": true,
      "scale": "m/s",
      "show": true,
      "spanGaps": undefined,
      "stroke": "#808080",
      "value": [Function],
      "width": 2,
    },
  ],
  "tzDate": [Function],
}
`;
